force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -43,7 +63,7 @@ public with sharing class NextStepFuture {
         insertBigDeal(bigDeal);
 
         UserRole[] roles = [SELECT Id, Name FROM UserRole WHERE Id = :UserInfo.getUserRoleId() OR Name = 'CRO' LIMIT 1];
-        updateUserRoleName(roles);
+       updateUserRoleName(roles);
```

Comment:
This is the test that is still failing. One of the strategies to avoid mixed DML is to move some of the processing to @future method. Can you refactor the code to do a portion of it as a future method?